### PR TITLE
Certificate pinning - establish TSL connection with DCC

### PR DESCRIPTION
**Pull request by alexmoroianu**

### CODE REVIEW

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/client/CloudDccFeignHttpClientProvider.java [hunk 1]
```diff
@@ -48,7 +61,24 @@ public Client createFeignClient() {
...
+  }
+
+  private SSLContext getSslContext(File trustStorePath, String trustStorePass) {
+    logger.info("Instantiating SSL context with keystore: " + trustStorePath.getName());
+    try {
+      return SSLContextBuilder.create().loadTrustMaterial(trustStorePath,
+              emptyCharrArrayIfNull(trustStorePass))
+          .build();
+    } catch (Exception e) {
+      logger.error("Problem on creating SSL context with key: " + trustStorePath.getName(), e);
```
Comment (roesslerj) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/client/CloudDccFeignHttpClientProvider.java:75]: I would change this a bit, as "key" might be misleading here... how about
"Problem creating SSL context with truststore "...

**Review by roesslerj — CHANGES_REQUESTED** [review]

**Comment by roesslerj** [comment]
Also, didn't you want to add a little script showing how to create the truststore? :-)

**Comment by alexmoroianu** [comment]
> Also, didn't you want to add a little script showing how to create the truststore? :-)

**Review by roesslerj — APPROVED** [review]

File: common/shared/src/test/java/app/coronawarn/server/common/shared/util/CwaStringTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import org.junit.jupiter.api.Test;
+
+class CwaStringTest {
+  private String STRING_TEST = "test me";
+  private String EMPTY_STRING = "";
+
+  @Test
+  void shouldConvertStringToCharArray() {
+    assertThat(String.valueOf(emptyCharrArrayIfNull(STRING_TEST))).isEqualTo(STRING_TEST);
+    assertThat(emptyCharrArrayIfNull(EMPTY_STRING)).isEmpty();
```
Comment (hilmarf) [inline common/shared/src/test/java/app/coronawarn/server/common/shared/util/CwaStringTest.java:24]:
```suggestion
    assertThat(emptyCharrArrayIfNull(EMPTY_STRING)).isEmpty();
    assertThat(emptyCharrArrayIfNull(null)).isEmpty();
```

File: common/shared/src/test/java/app/coronawarn/server/common/shared/util/CwaStringTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+
+
```
Comment (hilmarf) [inline common/shared/src/test/java/app/coronawarn/server/common/shared/util/CwaStringTest.java:2]:
```suggestion
```

**Review by hilmarf — APPROVED** [review]

**Pull request merged**